credential/credential.go

```diff
@@ -0,0 +1,163 @@
+/*
+ * Nuts node
+ * Copyright (C) 2021 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package credential
+
+import (
+	"encoding/json"
+	"fmt"
+	"path"
+	"time"
+
+	"github.com/nuts-foundation/go-did"
+	"github.com/nuts-foundation/go-leia"
+	"github.com/nuts-foundation/nuts-node/core"
+	"github.com/nuts-foundation/nuts-node/credential/concept"
+)
+
+// NewCredentialInstance creates a new credential instance with default config and empty concept registry
+func NewCredentialInstance() Credential {
+	return &credential{
+		config:   DefaultConfig(),
+		registry: concept.NewRegistry(),
+	}
+}
+
+type credential struct {
+	registry concept.Registry
+	config   Config
+	store    leia.Store
+}
+
+func (c *credential) Registry() concept.Registry {
+	return c.registry
+}
+
+func (c *credential) Reader() VCReader {
+	return c
+}
+
+func (c *credential) Configure(config core.ServerConfig) error {
+	var err error
+	fsPath := path.Join(config.Datadir, "credentials", "credentials.db")
```

Comment:
if we rename `credential` to `vcr` (or vcs) I'd name the directory `vcr` as well rather than `credentials`, because that's how the other engines do it. The database file name is fine though since it really stores the credentials.